scout/server/utils.py

```diff
@@ -227,6 +232,49 @@ def user_institutes(store, login_user):
     return institutes
 
 
+def is_token_expired(token: dict) -> bool:
+    """Check if the access token is expired."""
+    if "expires_at" not in token:
+        return True  # safer to assume expired
+    return token["expires_at"] <= time.time()
+
+
+def get_token_endpoint(discovery_url: str) -> Optional[str]:
+    """Returns the token endpoint for an OIDC provider."""
+
+    resp = requests.get(discovery_url)
+    resp.raise_for_status()
+    return resp.json()["token_endpoint"]
+
+
+def refresh_access_token_if_needed() -> None:
```

Comment:
Access token validity is custom, but it gets refreshed using refresh token when you open the case